### DISCUSSION TITLE
When will .NET Maui be feature complete?

### DISCUSSION THREAD

**Discussion by JohnHDev** [dotnet/maui#3854] [General]
We have started investigative work migrating apps to Maui, and have found features that are not yet complete. Looking at the status page: https://github.com/dotnet/maui/wiki/status we can see that Label.FormattedText is pending and we make considerable use of that.
When might FormattedText be completed? Is there a preview version when you expect all features to be complete (excluding bug fixing of course).

Thanks
John

**Comment by nogginbox** [comment]
According to the [Roadmap](https://github.com/dotnet/maui/wiki/Roadmap) Q1 2022, but I don't know if they are on track for this.

**Reply by JohnHDev** [reply]
Thanks NogginBox, Im aware of that but was looking for something a little more in-depth etc.

Ive looked at the Maui code for Label.FormattedText, it appears to be mostly a lift and shift from XF, so not sure why it does nothing at this point (tested on iOS and Windows). The platform specific extension methods appear to be in place, although the iOS side has an attribute saying partially ported, I can guess what that means.

I guess we'll just have to wait and see...

**Reply by nogginbox** [reply]
I'm currently quite frustrated by this too, but maybe I should have learned my lesson as this always happen when I use pre-release stuff. If you play on the bleeding edge, then you're going to get cut.

I'm currently battling with setting a title view on a Shellview and have no idea if it's not working because I'm wrong or because it's not implemented yet.

I would like a more detailed roadmap, plus the status page is for the latest dev commit rather than the latest released preview version. See [this issue](https://github.com/dotnet/maui/issues/3699) I created.

**Reply by nogginbox** [reply]
P.S. If you're looking at the source on github, then what you're looking at might not be the same as the release you're using.

**Reply by JohnHDev** [reply]
True, the code may have been updated but not deployed (at least not in P11).
Bleeding edge is right, which is why Im not stamping my feet and complaining and I say the same to other people, this is all still pre-release etc.

**Reply by JohnHDev** [reply]
Ha! It appears I was looking at the compatibility code for Label extensions on iOS, this hasn't actually been implemented yet using the new handler design at all.
